tests/subsys/openthread/radio_test.c

```diff
@@ -0,0 +1,818 @@
+		      "Channel number not remembered.");
+
+	zassert_true(otPlatRadioIsEnabled(ot), "Radio reports as disabled.");
+}
+
+/**
+ * @brief Test address filtering
+ * Tests if short, extended address and PanID are correctly passed to the radio
+ * driver.
+ *
+ */
+static void address_test(void)
+{
+	const u16_t pan_id = 0xDEAD;
+	const u16_t short_add = 0xCAFE;
+	otExtAddress ieee_addr;
+
+	for (int i = 0; i < sizeof(ieee_addr.m8); i++) {
+		ieee_addr.m8[i] = 'a' + i;
+	}
+
+	ztest_expect_value(filter_mock, set, true);
+	ztest_expect_value(filter_mock, type, IEEE802154_FILTER_TYPE_PAN_ID);
+	ztest_expect_value(filter_mock, filter->pan_id, pan_id);
+	otPlatRadioSetPanId(ot, pan_id);
+
+	ztest_expect_value(filter_mock, set, true);
+	ztest_expect_value(filter_mock, type,
+			   IEEE802154_FILTER_TYPE_SHORT_ADDR);
+	ztest_expect_value(filter_mock, filter->short_addr, short_add);
+	otPlatRadioSetShortAddress(ot, short_add);
+
+	ztest_expect_value(filter_mock, set, true);
+	ztest_expect_value(filter_mock, type, IEEE802154_FILTER_TYPE_IEEE_ADDR);
+	ztest_expect_data(filter_mock, filter->ieee_addr, ieee_addr.m8);
+	otPlatRadioSetExtendedAddress(ot, &ieee_addr);
+}
+
+/**
+ * @brief Test received messages handling.
+ * Tests if received frames are properly passed to the OpenThread
+ *
+ */
+static void receive_test(void)
+{
+	struct net_pkt *packet;
+	struct net_buf *buf;
+	const u8_t channel = 21;
+	const s8_t power = -5;
+	const u8_t lqi = 240;
+	const s8_t rssi = -90;
+	u8_t len;
+
+	packet = net_pkt_alloc(K_NO_WAIT);
+	buf = net_pkt_get_reserve_tx_data(K_NO_WAIT);
+	net_pkt_append_buffer(packet, buf);
+
+	for (int i = 0; i < buf->size; i++) {
+		buf->data[i] = ('a' + i) & 0xFF;
+	}
+	len = buf->size - 3;
+	buf->len = len;
+
+	net_pkt_set_ieee802154_lqi(packet, lqi);
+	net_pkt_set_ieee802154_rssi(packet, rssi);
+	/* Length inc. CRC. */
+	otPlatRadioSetTransmitPower(ot, power);
+
+	ztest_returns_value(set_channel_mock, 0);
+	ztest_expect_value(set_channel_mock, channel, channel);
+	ztest_expect_value(set_txpower_mock, dbm, power);
+	ztest_expect_value(start_mock, dev, &radio);
+	otPlatRadioReceive(ot, channel);
+	/*
+	 * Not setting any expect values as nothing shall be called from
+	 * notify_new_rx_frame calling thread. OT functions can be called only
+	 * after semaphore for main thread is released.
+	 */
+	notify_new_rx_frame(packet);
+
+	ztest_expect_value(otPlatRadioReceiveDone, aError, OT_ERROR_NONE);
+	ztest_expect_value(otPlatRadioReceiveDone, aFrame->mChannel, channel);
+	ztest_expect_value(otPlatRadioReceiveDone, aFrame->mLength, len);
+	ztest_expect_data(otPlatRadioReceiveDone, aFrame->mPsdu, buf->data);
+	make_sure_sem_set(Z_TIMEOUT_MS(100));
+	platformRadioProcess(ot);
+}
+
+void test_main(void)
+{
+	platformRadioInit();
+
+	ztest_test_suite(
+		openthread_radio, ztest_unit_test(energy_scan_immediate_test),
+		ztest_unit_test(energy_scan_delayed_test),
+		ztest_unit_test(tx_test), ztest_unit_test(tx_power_test),
```

Comment:
Could we split the tests into one line each? I think it would be easier to follow what tests are available.